tests/deprecated_api/test_remove_1-8.py

```diff
@@ -608,3 +608,10 @@ def on_pretrain_routine_end(self, trainer, pl_module):
         match="The `Callback.on_pretrain_routine_end` hook has been deprecated in v1.6" " and will be removed in v1.8"
     ):
         trainer.fit(model)
+
+
+def test_v1_8_0_weights_save_path(tmpdir):
+    with pytest.deprecated_call(match=r"Setting `Trainer\(weights_save_path=\)` has been deprecated in v1.6"):
+        _ = Trainer(
+            weights_save_path=tmpdir,
+        )
```

Comment:
```suggestion
        _ = Trainer(weights_save_path=tmpdir)
```

Reply:
this test should also confirm that the modelcheckpoint callback dir path is still set to weight save path so we can refactor the other tests, right?

Reply:
I believe this is done in `test_all` below?

Reply:
what I'm trying to say is, we can make deprecation test here in test_remove_1-8.py where we do:

```py
trainer = Trainer(weight_save_path="a/b/c")
assert trainer.checkpoint_callback.dir_path == "a/b/c"
```

so that we can update other tests to use the new API / get rid of weight_save_path.

Reply:
@awaelchli I was under the impression that in the `test_remove_x-y.py` we generally want to test the deprecation warnings, and the functionality of the deprecated behavior should be tested in the same tests that we have been using  prior to deprecation. Following this logic, shouldn't we keep this test here which tests the functionality of weights_save_path? Is my understanding wrong of what we use the `test_remove_x-y.py` for?

https://github.com/PyTorchLightning/pytorch-lightning/blob/29d5afbda6ad64e59241950d0b94b19564c2710e/tests/loggers/test_all.py#L223